### ISSUE TITLE
model: Expose more information

### ISSUE THREAD

**Issue by pdxjohnny** [intel/dffml#1143]
We know we need to expose the following before stable API release:\n- Model type\n  - Classifier, Regressor, Cluster, NLP, etc. Need to enumerate all these we have right now\n- Model trained status. Boolean value which we need to know before attempted to assess accuracy or prediction

**Comment by programmer290399** [comment]
I think I can pick this up for this week

**Comment by programmer290399** [comment]
* What would be the correct defining the Model type. Shall we keep it as a property of the model that is set manually while defining the model class ?

* Or it should be a config property which is immutable, and we have a exhaustive set of types to choose from. 

* In both cases I would need  an exhaustive set of types for models. 

* Another way to deal with this could be to return the model object's class name. by making a `classmethod` to return the class name of the object.

**Comment by pdxjohnny** [comment]
> Or it should be a config property which is immutable, and we have a exhaustive set of types to choose from.

Config properties are for runtime specific values. Do we have any situations, or can we think of any, where we might want to set the usage at runtime? The only one I can think of off the top of my head is treating a classification model's output as if it was a regression model.

**Comment by pdxjohnny** [comment]
Putting it in the config, with a default value set, would be a convenient place for it.

**Comment by mHash1m** [comment]
Okay so, I'll talk here about enumerating the types.

From the types mentioned above, one may be a sub-type of the other... 
For example, there are nlp and cv models that can be classified under 'classifiers' and vice versa.

They could be classified into: 

- supervised and unsupervised
- classifiers, regressors, and clustering models.
- linear models, naive bayes, decision trees, ensemble methods, neural networks, etc.
- NLP, CV etc.
- and maybe more

Every NLP and CV model out there could be tagged with a type from each of the lists above.

Say, a dog classifier on Pytorch would be a classifier, a CV model, and a neural network model.
The approach really depends on the amount of detail we want to go into. 

Also, will we be using this to later filter out models in UI? If yes, we'd want to somehow tag each model in every way we can(having multiple tags) to later have more options for filtering.

**Comment by programmer290399** [comment]
Exactly @mHash1m !! We can differentiate between models in a lot of ways and we need to make sure that we're on the same page about this and thus we should definitely pick this up in next weekly sync. Until then I think I can make PR implementing  `Model trained/not-trained status` as it is pretty simple and there's no confusion around it as well.
